src/Reflection/ReflectionClass.php

```diff
@@ -236,7 +238,16 @@ public static function createFromNode(
      */
     public function getShortName() : string
     {
-        return $this->node->name;
+        if (!$this->isAnonymous()) {
+            return $this->node->name;
+        }
+
+        $fileName = $this->getFileName();
+        if ($fileName === null) {
```

Comment:
The `$fileName` can be empty only if you use `StringSourceLocator`. I hope I've solved it.

I throw an exception for evaled anonymous class currenly because it's little complicated to solve it. The PhpParser returns this:

```
array(1) {
  [0]=>
  object(PhpParser\Node\Expr\Eval_)#32 (2) {
    ["expr"]=>
    object(PhpParser\Node\Scalar\String_)#31 (2) {
      ["value"]=>
      string(20) "$a = new class() {};"
      ["attributes":protected]=>
      array(3) {
        ["startLine"]=>
        int(3)
        ["endLine"]=>
        int(3)
        ["kind"]=>
        int(1)
      }
    }
    ["attributes":protected]=>
    array(2) {
      ["startLine"]=>
      int(3)
      ["endLine"]=>
      int(3)
    }
  }
}
```
 
It can be probably solved but I have to figure out how to solve start and end line numbers. I think it would be better to do it in another PR.